### PR TITLE
Trigger groups!!

### PR DESCRIPTION
**Pull request by kadst43**

### CODE REVIEW

**Comment by kadst43** [comment]
@AmbientOSX

Implements trigger groups.  Also cleans up some of the places we were using the blackboard to prevent repetition, replacing them with the doOnce clause.

Tested, and this does fix the issue where both success and failure fire simultaneously.

File: source/scenario/globalScenarioTriggers.vwf.yaml [hunk 1]
```diff
@@ -127,14 +118,13 @@ properties:
         - musicFailure
 
     failOnCollision:
+      group: successOrFailure
+      priority: 2
       triggerCondition:
       - and:
```
Comment (AmbientOSX) [inline source/scenario/globalScenarioTriggers.vwf.yaml]: No need for an AND here

File: source/scenario/globalScenarioTriggers.vwf.yaml [hunk 2]
```diff
@@ -178,14 +168,13 @@ properties:
           - ALVO32_Rover
 
     failOnBatteryDepleted:
+      group: successOrFailure
+      priority: -2
       triggerCondition:
       - and:
```
Comment (AmbientOSX) [inline source/scenario/globalScenarioTriggers.vwf.yaml]: No AND needed

**Comment by AmbientOSX** [comment]
Rest looks :+1:

**Pull request merged**